### DISCUSSION TITLE
Server rendered collapsed panel based on route

### DISCUSSION THREAD

**Discussion by Andi365** [bvaughn/react-resizable-panels#275] [Q&A] [answered]
Hi!

I use Nextjs App router and need to determine if the panel should be collapsed on a per-route basis. 

Here is my problem:

On my route /customers I need the panel to be collapsed=false, but on the route /customers/[customerId] I need the panel to be collapsed=true by default. My desired outcome would be to render this on the server, where I pass the collapsed on a per-route basis. My PanelGroup is contained in the base path /customers. I tried to alter the cookies like so on my /customers/[customerId] route:

`cookies().set("react-resizable-panels:collapsed", "false")`, but nextjs doesnt allow this: 
![image](https://github.com/bvaughn/react-resizable-panels/assets/43875035/a8548fa1-24da-4d7d-b224-8259d4ec6017)

another approach could be to alter this with a provider or use a useeffect in a client component on mount to set the cookie, but this will lose the non-flicker ability by server rendering.

Is there any way I can make this work that I overlook at the moment?

Thank you in advance!

**Comment by bvaughn** [comment, marked as answer]
Seems like rendering `defaultSize={0}` (replace 0 with whatever your `collapsedSize` is if not 0) should work. Does this not work, for some reason?
